snuba/consumer.py

```diff
@@ -24,9 +27,15 @@ def __init__(self, clickhouse, dataset, producer, replacements_topic, metrics=No
         self.metrics = metrics
 
     def process_message(self, message):
+        # TODO: consider moving this inside the processor so we can do a quick
+        # processing of messages we want to filter out without fully parsing the
+        # json.
         value = json.loads(message.value())
-
-        processed = self.__dataset.get_processor().process_message(value)
+        metadata = {
+            KAFKA_MESSAGE_OFFSET_KEY: message.offset(),
+            KAFKA_PARTITION_KEY: message.partition(),
+        }
```

Comment:
Why use a dictionary here rather than something with a more enforced structure like a namedtuple?

Reply:
makes sense, will turn to namedtuple